modules/repository/temp.go

```diff
@@ -13,30 +14,36 @@ import (
 	"code.gitea.io/gitea/modules/util"
 )
 
-// LocalCopyPath returns the local repository temporary copy path.
-func LocalCopyPath() string {
-	if filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
-		return setting.Repository.Local.LocalCopyPath
+// localCopyPath returns the local repository temporary copy path.
+func localCopyPath() string {
+	if setting.Repository.Local.LocalCopyPath == "" {
+		return filepath.Join(setting.TempPath, "local-repo")
+	} else if !filepath.IsAbs(setting.Repository.Local.LocalCopyPath) {
+		return filepath.Join(setting.TempPath, setting.Repository.Local.LocalCopyPath)
 	}
-	return filepath.Join(setting.AppDataPath, setting.Repository.Local.LocalCopyPath)
```

Comment:
The temp local copy path now moved to temp path rather than `setting.AppDataPath`.